### ISSUE TITLE
Error: EMFILE, too many open files '/proc/14960/stat

### ISSUE THREAD

**Issue by wwwy3y3** [NodeFly/NodeFly#17]
Hi

I ran into this error

```
nodefly error: { [Error: EMFILE, too many open files '/proc/14960/stat']
 errno: 20,
 code: 'EMFILE',
 path: '/proc/14960/stat', syscall: 'open' } 
 Error: EMFILE, too many open files '/proc/14960/stat'
   at Object.fs.openSync (fs.js:338:18)
   at Object.fs.readFileSync (fs.js:182:15)
   at Object.exports.cpuutil    (/home/ubuntu/prod/node_modules/nodefly/lib/cpuinfo.js:22:22)
   at collect (/home/ubuntu/prod/node_modules/nodefly/lib/info.js:78:23)
   at Timer.<anonymous> (/home/ubuntu/prod/node_modules/nodefly/lib/info.js:41:4)
   at Timer.exports.setInterval.timer.ontimeout (timers.js:234:14)
```

what does that mean?

seem to happen only on high load

**Comment by groundwater** [comment]
This was an unhandled edge case in our agent, I have fixed it in version 0.1.26. The agent will no longer throw an exception if it cannot open read the stat file.

It is unlikely that our agent is using up the file handles, is your application opening individual files each request? You may want to also boost the number of allowable file handles:
- http://www.cyberciti.biz/faq/linux-increase-the-maximum-number-of-open-files/

**Comment by wwwy3y3** [comment]
turn out I'm not using `NODE_ENV=production` in express
I will also notice the ulimit!
thanks!
